internal/app/wrapperd/main.go

```diff
@@ -0,0 +1,98 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package wrapperd provides a wrapper for running services.
+package wrapperd
+
+import (
+	"flag"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/containerd/cgroups"
+	cgroupsv2 "github.com/containerd/cgroups/v2"
+	"github.com/containerd/containerd/sys"
+	"github.com/siderolabs/gen/slices"
+	"golang.org/x/sys/unix"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+)
+
+var (
+	name        string
+	droppedCaps string
+	cgroupPath  string
+	oomScore    int
+)
+
+// Main is the entrypoint into /sbin/wrapperd.
+// nolint: gocyclo
+func Main() {
+	flag.StringVar(&name, "name", "", "process name")
+	flag.StringVar(&droppedCaps, "dropped-caps", "", "comma-separated list of capabilities to drop")
+	flag.StringVar(&cgroupPath, "cgroup-path", "", "cgroup path to use")
+	flag.IntVar(&oomScore, "oom-score", 0, "oom score to set")
+	flag.Parse()
+
+	if droppedCaps != "" {
```

Comment:
should we drop capabilities last, as we might not be able to adjust OOM score/enter cgroup with lowered caps?

Reply:
good point, I'll move that